### PR TITLE
chore: improve master logging

### PR DESCRIPTION
**Pull request by neilconway**
Two main changes:

  * Reduce gRPC log level from INFO to WARN. gRPC INFO log messages are
    almost universally about cryptic implementation details, such as

      pickfirstBalancer: HandleSubConnStateChange: 0xc0006a2010, {READY <nil>}

  * Squelch a log message ("initializing endpoints for agents") that does
    not seem useful.

Along the way, fix various typos and improve log message consistency.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

**Comment by neilconway** [comment]
@stoksc Not sure offhand if it is okay to create multiple logrus loggers...

**Review by stoksc — APPROVED** [review]
lgtm

**Comment by stoksc** [comment]
from an example in the logrus docs:
> // Create a new instance of the logger. You can have any number of instances.

I just assumed, based on the thinking `logrus.New()` would be a weird function to have otherwise.

**Comment by stoksc** [comment]
Huh wait, maybe that's multiple instances of the same logger.. I guess I'll read better.

File: master/internal/grpcutil/api.go [hunk 1]
```diff
@@ -26,22 +26,25 @@ const jsonPretty = "application/json+pretty"
 
 // NewGRPCServer creates a Determined gRPC service.
 func NewGRPCServer(db *db.PgDB, srv proto.DeterminedServer) *grpc.Server {
-	logger := logrus.NewEntry(logrus.StandardLogger())
+	logger := logrus.New()
```
Comment (stoksc) [inline master/internal/grpcutil/api.go:29]: I spent some time reading the logrus code - afaict it seems totally fine to have two logrus loggers but I'm always happy to be surprised..

Comment (neilconway) [reply master/internal/grpcutil/api.go:29]: Yep, that is my reading as well -- assuming the underlying `io.Writer` is safe to write to concurrently, which I think should be the case here with `os.Stderr`.

**Pull request merged**